src/main/java/StringUtils.java

```diff
@@ -0,0 +1,20 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringUtils {
+    public static String[] splitString(String text) {
+        String key = ",|:";
+        Matcher m = findSplitKey(text);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            key += "|";
+            key += customDelimiter;
+            return m.group(2).split(key);
+        }
+        return text.split(key);
+    }
+
+    private static Matcher findSplitKey(String text) {
+        return Pattern.compile("//(.)\n(.*)").matcher(text);
```

Comment:
Pattern은 비싼 객체인데, pattern을 아래와 같이 상수로 정의해서 사용하는 건 어떨까요?

```suggestion
    private static final Pattern SPLIT_KEY_PATTERN = Pattern.compile("//(.)\n(.*)");
```

아래 링크를 한번 참고해보시면 좋을것 같네요!

성능을 개선하려면 Pattern 객체를 만들어 재사용하는 것이 좋다.

```java
public class RomanNumber {

    private static final Pattern ROMAN = Pattern.compile("^(?=.)M*(C[MD]|D?C{0,3})(X[CL]|L?X{0,3})(I[XV]|V?I{0,3})$");

    static boolean isRomanNumeral(String s) {
        return ROMAN.matcher(s).matches();
    }
}
```

https://github.com/keesun/study/blob/master/effective-java/item6.md